conbench/entities/benchmark_result.py

```diff
@@ -1128,33 +1087,58 @@ class _BenchmarkResultCreateSchema(marshmallow.Schema):
         metadata={
             "description": conbench.util.dedent_rejoin(
                 """
-                Identifier for a Run (required). This can be the ID of a known
-                Run (as returned by /api/runs) or a new ID in which case a new
-                Run entity is created in the database.
+                Unique identifier for a "run" of benchmarks (typically a single run of a
+                CI pipeline) on a single commit and hardware. Required.
+
+                The Conbench UI and API make several assumptions that all benchmark
+                results with the same `run_id` share the same `run_tags`, `run_reason`,
+                hardware, and commit. There is no technical enforcement of this on the
+                server side, so some behavior may not work as intended if this
+                assumption is broken by the client.
                 """
             )
         },
     )
-    run_name = marshmallow.fields.String(
+    run_tags = marshmallow.fields.Dict(
         required=False,
         metadata={
             "description": conbench.util.dedent_rejoin(
                 """
-                Name for the Run (optional, does not need to be unique). Can be
-                useful for implementing a custom naming convention. For
-                organizing your benchmarks, and for enhanced search &
-                discoverability. Ignored when Run was previously created.
+                An optional mapping of arbitrary keys and values that describe the CI
+                run. These are used to group and filter runs in the UI and API. Do not
+                include `run_reason` here; it should be provided below.
+
+                For legacy reasons, if `run_name` is given when POSTing a benchmark
+                result, it will be added to `run_tags` automatically under the `name`
+                key. This will be its new permanent home.
```

Comment:
`run_name` frustrates me a bit :)  https://github.com/conbench/conbench/issues/607

> The linked issues/PRs show that run_name creates work, while the value is still unclear to me.

but yeah, in this patch we still have to drag this on.

Reply:
I considered removing it completely but now it's tied up in the history endpoint:

https://github.com/conbench/conbench/blob/42ef379825e0bb8d7cedc00f0d143e84f102c60b/conbench/entities/history.py#L110

I still think there's value in cleaning that up in #607.